build: remove NUKE.build

BrycensRanch opened the pull request:
# Preface

The purpose of this PR is to remove the dependency on NUKE.Build. 

Ever since SnapX [started using slnx](https://github.com/BrycensRanch/SnapX/commit/575a7adbd01bfa1dd115ad07011786cde256ddc3) for a clearer format than `sln`, the NUKE.Build tooling has **not** kept up. 

That commit led to a massive build failure due to a lack of support. Upon looking further in the NUKE.Build GitHub issues, I saw [this](https://github.com/nuke-build/nuke/issues/1375).

 For me, this was the first time I had seen an OSS maintainer quoting Twitter as a reason to close a legitimate feature request. Since then, I have been silently looking for a way out of using NUKE.build as the build system. 

I was considering using `GNU Make`; however, it's far removed from the .NET ecosystem. So I chose less integrated solutions in favor of more control. That's where [Bullseye](https://www.nuget.org/packages/Bullseye/) & [SimpleExec](https://www.nuget.org/packages/SimpleExec) come in.


## Notes

This PR also removes the dependency on Bash, and instead leans on standard POSIX scripting for `build.sh`. However, the target `install` is still being worked on.


This PR has been upgraded from optional to urgent due to the [NUKE.Build GitHub repository being archived](https://github.com/nuke-build/nuke). Unfortunately with some .NET libraries going commercial as of late, I can only assume the NUKE.Build maintainers thought it was a great idea.

### Additional plans

When .NET 10 releases, we will investigate if [this feature will assist](https://youtu.be/98MizuB7i-w?t=186).

Also, the deprecated `sln` file will be deleted when .NET 10 releases.

BrycensRanch committed:
build: remove NUKE.build

NUKE.build lacks support for slnx. So I've opted to remove it for a more simpler solution.

BrycensRanch committed:
docs: remove mentions of NUKE

BrycensRanch committed:
chore: remove unnecessary build.cmd

Windows directions specifically say to use build.ps1

BrycensRanch committed:
Merge remote-tracking branch 'origin/develop' into build/remove-nuke-build

# Conflicts:
#	build/Build.cs
#	build/_build.csproj

ok-coder1 commented:
![Screen Shot 2025-05-20 at 8 07 51 PM](https://github.com/user-attachments/assets/97b42877-72d8-474c-abeb-bfd113b14b1d)


**_temporarily archived_**. It's not going to be archived permanently.

BrycensRanch commented:
> ![Screen Shot 2025-05-20 at 8 07 51 PM](https://github.com/user-attachments/assets/97b42877-72d8-474c-abeb-bfd113b14b1d)
> 
> 
> 
> 
> 
> **_temporarily archived_**. It's not going to be archived permanently.

Temporarily archived means they're going to do something stupid and want attention, thus the temporary archival. Their repository had plenty of commits as of late. 

It's a move that I want no part in. Normal open source maintainers do not do this.

BrycensRanch committed:
build: remove dependency on bash

BrycensRanch committed:
build: add support for build.ps1 to be ran via git bash

BrycensRanch committed:
Merge remote-tracking branch 'origin/develop' into build/remove-nuke-build

BrycensRanch committed:
build: add missing build arguments

Not everything, the only arguments that are still missing are the ones for the install target.

BrycensRanch committed:
ci(build): use slnx entirely

BrycensRanch committed:
ci(build): use new build proj name

BrycensRanch committed:
build: use proper System.Commandline version

BrycensRanch committed:
build: finish build & install steps

I'm losing sleep.

BrycensRanch committed:
build: make compatible with current packaging

BrycensRanch committed:
build: add back sln file until .NET 10 drops

If I didn't, most distros dotnet-sdk package wouldn't compatible anymore.

BrycensRanch committed:
ci(build): backport commit `8ee45c1cf27f8cb541e7a26b896d0a6842cb1247`
build: always install latest .NET version

I am referring to commit https://github.com/BrycensRanch/SnapX/commit/8ee45c1cf27f8cb541e7a26b896d0a6842cb1247

BrycensRanch committed:
build: use POSIX pipefail alternative

BrycensRanch committed:
build: safely handle unset BASH_SOURCE to prevent set -u errors

BrycensRanch committed:
ci(build): order dotnet format arguments properly

BrycensRanch commented:
Great work, Brycen. Your pull request is merged.

matkoch commented:
@BrycensRanch if I may also take out my crystal ball for a moment – next Monday you will feel very silly.
